### PR TITLE
Introduce JRuby.md

### PR DESCRIPTION
**Pull request by fzakaria**
Introduce a new markdown file describing using Sorbet
with JRuby.

It covers some gotchas I came across as I introduced Sorbet
into our codebase.

### CODE REVIEW

**Comment by fzakaria** [comment]
tagging @DarkDimius  & @jasnow

File: JRuby.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+## JRuby
```
Comment (jez) [inline JRuby.md]: This file should not live at the top-level. If anything it should be in `docs/`

File: JRuby.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+> The Ruby Programming Language on the JVM
+
+This document serves as a getting-started guide for those wishing to use
+[Sorbet](https://sorbet.org) with [JRuby](https://www.jruby.org/).
+
+The use of Sorbet on the JVM is not **officially** supported by the Sorbet
```
Comment (jez) [inline JRuby.md]:
```suggestion
The use of Sorbet on the JVM is **NOT SUPPORTED** by the Sorbet
```

File: JRuby.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+development team however this document services as a getting started guide
+for those who wish to trot on a unsupported path.
```
Comment (jez) [inline JRuby.md]: Can you add a note here that being not supported means bug reports that only reproduce on JRuby are likely to not get fixed or potentially even closed.

File: JRuby.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+### Performance
+
+TBD
+
+### Gotchas
+
+#### RBI for Java Code
```
Comment (jez) [inline JRuby.md]: Non-blocking request:

Please use the same nesting structure as `docs/internals.md`. Namely: Start with `#` for the title of the doc, and only use `##` / `###` for sections and subsections (and no sub-subsections, i.e., no `####`)

File: JRuby.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+The `srb` tool can auto-generate RBI files for your gems however it cannot
+do so for your Java code :(
```
Comment (jez) [inline JRuby.md]: Non-blocking request:

- Can we make the tone more neutral? i.e., no `:(`, no first person.

File: JRuby.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    |   |---------- java_lang.rbi # this is the standard library
+    |   |---------- org_eclipse_jgit.rbi
+    |   |---------- org_apache_logging_log4j.rbi
+    |-------------  org_slf4j.rbi
+```
+
+Basically, I take the _maven coordinates_ for a given java packet (replaced with
+underscores) and hand-create an RBI file for it.
+
+You can put as much as you need depending on the `#typed` sigils you are aiming
```
Comment (jez) [inline JRuby.md]:
```suggestion
You can put as much as you need depending on the `# typed:` sigils you are aiming
```

File: JRuby.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+You might have added `sorbet` to your Gemfile but noticed that sorbet-static
+was not included and that `srb` failed!
+
+This is because [at the moment](https://github.com/sorbet/sorbet/pull/2254) the
+Sorbet team does not publish a `sorbet-static` gem for the Java platform.
+
+The workaround is to fetch the Gem manually for your arch and install it.
+
+For instance on OSX you would do:
```
Comment (jez) [inline JRuby.md]:
```suggestion
For instance on macOS you would do:
```

File: JRuby.md (hunk 3 above)
Comment (fzakaria) [reply JRuby.md]: Will add

File: JRuby.md (hunk 5 above)
Comment (fzakaria) [reply JRuby.md]: Will comb over

File: JRuby.md (hunk 1 above)
Comment (fzakaria) [reply JRuby.md]: Fixed

**Comment by fzakaria** [comment]
I gave the document a "treatment" as you suggested @jez

**Pull request merged**